src/components/AddSemesterModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useContext, useState } from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import DefaultModal from './DefaultModal';
+import TextInput from './TextInput';
+import Logo from '../molecules/Logo';
+
+const Body = styled.div`
+	height: 100%;
+	width: 100%;
+	display: grid;
+	grid-template-rows: 1fr 4fr 2fr;
+	grid-template-areas:
+		'desc'
+		'logo'
+		'input';
+	justify-items: center;
+`;
+const Desc = styled.p`
+	grid-area: desc;
+	font-size: ${({ theme }) => theme.fontSizes.xs};
+	margin: 1em 0 0 0.2em;
+`;
+
+const BodyLogo = styled(Logo)`
+	grid-area: logo;
+`;
+
+const InputField = styled.div`
+	grid-area: input;
+	padding: 0 0 0.5em 0;
+	margin: 0 2.2em 0 2em;
```

Comment:
These are very specific margins and paddings. It makes me wonder if something isn't quite flowing right. How does this look on different screen sizes? Are these the same margins in another component?

Reply:
These were the same margins/paddings from other modal components that I used as reference. Just tested removing both properties and the difference is trivial -- the input field shifts every so slightly to the right (0.2em!)
I noticed that the description div (Desc) also had specific margins, so tested removing those. 
I don't think they are required so I removed all.